examples/generic.go

```diff
@@ -0,0 +1,61 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+
+	"github.com/looplab/fsm"
+)
+
+type MyEvent string
+type MyState string
+
+const (
+	Close MyEvent = "close"
+	Open  MyEvent = "open"
+	Any   MyEvent = ""
+
+	IsClosed MyState = "closed"
+	IsOpen   MyState = "open"
+)
+
+func main() {
+	fsm := fsm.New(
+		IsClosed,
+		fsm.Transitions[MyEvent, MyState]{
+			{Event: Open, Src: []MyState{IsClosed}, Dst: IsOpen},
+			{Event: Close, Src: []MyState{IsOpen}, Dst: IsClosed},
+		},
+		fsm.Callbacks[MyEvent, MyState]{
```

Comment:
Can more types be inferred here?

Reply:
What do you mean with this ?

Reply:
I haven’t used generics in Go so much, it just felt there was some repetition in specifying the types. Thought maybe more types could be inferred, but that was more of an open question.

Reply:
Yes, type inference is the most complicated part in the generics of the go compiler, i tried making this a bit lighter by have a fluent interface for the FSM creation, something like:

```go
fsm := fsm.New(initialState).WithTransitions(transitions).WithCallbacks(callbacks)
```
Methods usually make inference easier for the compiler, but it didnt work out.